src/pybind/mgr/orchestrator.py

```diff
@@ -447,38 +447,80 @@ def to_json(self):
         return {k: v for (k, v) in out.items() if v is not None}
 
 
+class DeviceSelection(object):
+    def __init__(self, paths=None, id_model=None, size=None, rotates=None, count=None):
+        # type: (List[str], str, int, bool, int) -> ()
+        """
+        ephemeral drive group device specification
+        :param paths: abs paths to the devices.
+        :param id_model: A wildcard string. e.g: "SDD*"
+        :param size: Size specification of format LOW:HIGH.
+            Can also take the the form :HIGH, LOW:
+            or an exact value (as ceph-volume inventory reports)
+        :param rotates: is the drive rotating or not
+        :param count: if this is present limit the number of drives to this number.
+
+        Any attributes (even none) can be included in the device
+        specification structure.
```

Comment:
Is for that reason, that i suggested to use a "target" like attribute where a "linux file pattern" can be used to select a group of hosts ( "*" , "server-0?, ...)
I think that the orchestrator module can receive the whole  "drive group" , use the inventory to select the right hosts, and make the optimized number of operations with the external orchestrator to create the osd's

Reply:
@tserong  :+1: 
Also I think this could use a more explicit specification. E.g. how do the arguments interact with each other (I assume the resulting drives fulfil all selectors?).

Reply:
How is DeepSea going to implement this?

Reply:
@tserong I hesitated to add an "all" concept to the drive group, until c-v supports that.

Reply:
* @jan--f : Added a `TODO`
* @tserong: paths and other parameters are mutually exclusive for now.